thrift/transport.go

```diff
@@ -49,6 +54,37 @@ func (t *readWriterTransport) Close() error {
 	return nil
 }
 
+func (t *readWriterTransport) ReadByte() (byte, error) {
+	v := t.readBuf[0:1]
+
+	var n int
+	var err error
+
+	for {
+		n, err = t.Read(v)
```

Comment:
Is it actually possible for the transport to read zero bytes and not return an error?